### PR TITLE
Decompile N64 anti-piracy checks in overlays

### PR DESCRIPTION
**Pull request by cadmic**
All 3 checks seemingly use a different mechanism (for extra security?):
* In z_bg_zg.c, it checks if the value at RAM address 0x800002E8 is set to 0xC86E2000. I'm not really sure what sets this address, maybe IPL3? If the check fails, the default action func for the bars in Ganon's Tower escape does nothing, so they will not open when their switch flags are set.
* In z_en_zl2.c, it checks that the value of the IPL variable `osCicId` is 6105. If the check fails, Zelda gets a new hairdo due to an extra `Matrix_Scale` before drawing the hair (I assume).
* In z_fishing.c, it checks if some BSS variable in cic610c.c is set to the value 0xAD090010. If the check fails, `sReelLock` is set to 1 which means the line can't be reeled in if a fish is caught, and caught fish will always let go after 50 frames.

### CODE REVIEW

**Review by fig02 — APPROVED** [review]

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 1]
```diff
@@ -71,6 +71,13 @@ s32 func_808C0CC8(BgZg* this) {
 }
 
 void func_808C0CD4(BgZg* this, PlayState* play) {
+#if PLATFORM_N64
+    // Anti-piracy check, bars will not open if the check fails
+    if (IO_READ(0x800002E8) != 0xC86E2000) {
```
Comment (Thar0) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c:76]:
```suggestion
    if (IO_READ(0x000002E8) != 0xC86E2000) {
```
When using `IO_READ` it's better to leave the address as a physical address

For some context on when this is written, it's written when IPL3 copies itself to RDRAM after RDRAM has been initialized. Specifically, this is an instruction from some embedded rsp code at offset 0x7F8 into IPL3 (`0xC86E2000` disassembles to `lqv $v14[0], ($3)`)

Comment (fig02) [reply src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c:76]: could be cool to comment on those specifics in the code, just for completeness sake

Comment (cadmic) [reply src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: Thanks! I added this as a comment (essentially verbatim)

**Comment by Dragorn421** [comment]
I have a comment on "anti piracy"

These anti-piracy checks are no longer of any relevance today: they merely weed out bad emulation (specifically, bad emulation of the CIC). An example being Nintendo's GC emulator, hence why the checks were removed in GC versions.
These checks were effective at the time of releasing OoT64 because then emulators were all "bad", the CIC was new-ish, and in particular the CIC6105 was new as far as I know.

Today emulators bypass these "anti piracy" checks merely by emulating the system correctly. It's clear the intent in 1998 was for these checks to be for anti-piracy, but they no longer have that role today (they're meaningless now)

I'm saying this to defend against any "decomp is piracy" accusations. In today's context these checks shall not be seen as anti-piracy measures. It would be nice to find a way to refer to these checks as the legacy checks they are. Maybe "anti piracy checks of the time", "1998 anti piracy checks", "legacy anti piracy checks" ...

**Comment by cadmic** [comment]
I think the main purpose was actually to guard against repro carts without the CIC6105 chip. It's trivial to bypass with emulation, but it's harder to bypass if you're trying to run on an unmodified console (that is, with an unmodified IPL3).

I think the term "anti-piracy" is fine, it pretty clearly refers to the code itself implementing some kind of anti-piracy check, not some way to prevent decompilation.

**Review by Thar0 — APPROVED** [review]

**Comment by fig02** [comment]
agree that it doesnt really need any more clarification

**Pull request merged**